assets/javascripts/lib/ajax.coffee

```diff
@@ -74,45 +76,13 @@ applyHeaders = (xhr, options) ->
 
   if options.dataType
     options.headers['Accept'] = MIME_TYPES[options.dataType] or options.dataType
+  return options.headers
 
-  for key, value of options.headers
-    xhr.setRequestHeader(key, value)
+onSuccess = (data, options) ->
+  options.success?.call options.context, data, options
   return
 
-onComplete = (xhr, options) ->
-  if 200 <= xhr.status < 300
-    if (response = parseResponse(xhr, options))?
-      onSuccess response, xhr, options
-    else
-      onError 'invalid', xhr, options
-  else
-    onError 'error', xhr, options
+onError = (type, error, options) ->
+  options.error?.call options.context, type, error, options
   return
 
-onSuccess = (response, xhr, options) ->
-  options.success?.call options.context, response, xhr, options
-  return
-
-onError = (type, xhr, options) ->
-  options.error?.call options.context, type, xhr, options
-  return
-
-onTimeout = (xhr, options) ->
-  xhr.abort()
-  onError 'timeout', xhr, options
-  return
-
-abort = (xhr) ->
-  clearTimeout(xhr.timer)
-  xhr.onreadystatechange = null
-  xhr.abort()
-  return
```

Comment:
Here's one usage of abort:

- https://github.com/freeCodeCamp/devdocs/blob/93a42ccbcd69243b4aaa1dba292724aa625a844b/assets/javascripts/views/content/entry_page.coffee#L112-L114